### ISSUE TITLE
Build instructions missing "go get" and setting GOPATH?

### ISSUE THREAD

**Issue by Flowdalic** [miekg/mmark#104]
The build instructions currently only state

```
% cd mmark
% go build
```

while I had to

``` bash
cd mmake
export GOPATH=$(pwd)
go get -d -t -v ./...
go build
```

to make it work

**Comment by miekg** [comment]
I'm in favor of removing those instructions as they are no different than any go project.

**Comment by Flowdalic** [comment]
For someone like me, who is not used how go projects are typically build, those instructions are a life saver. Additionally/Alternatively I enjoy a Makefile so that I can build a project just by "git clone mmak && cd mmark && make". Yesterday, I had to look into the `.travis.yml` file to figure out what's missing.

**Comment by miekg** [comment]
Makefile sounds good, but we need to be smart about gopath

**Comment by runeimp** [comment]
Yeah, the docs are not real helpful and being a journeyman Go programmer shouldn't be a requirement just to get started. Following is what I experience on my Mac. A cautionary tale, if you will.

``` bash
iMac-27:repos runeimp$ git clone git@github.com:miekg/mmark.git
iMac-27:repos runeimp$ cd mmark/
iMac-27:mmark runeimp$ go build
toml.go:7:2: cannot find package "github.com/BurntSushi/toml" in any of:
	/usr/local/opt/go/libexec/src/github.com/BurntSushi/toml (from $GOROOT)
	/Users/runeimp/dev/go/lib/src/github.com/BurntSushi/toml (from $GOPATH)
	/Users/runeimp/dev/go/runeimp/src/github.com/BurntSushi/toml
iMac-27:mmark runeimp$ cd mmark/
iMac-27:mmark runeimp$ go build
main.go:21:2: cannot find package "github.com/miekg/mmark" in any of:
	/usr/local/opt/go/libexec/src/github.com/miekg/mmark (from $GOROOT)
	/Users/runeimp/dev/go/lib/src/github.com/miekg/mmark (from $GOPATH)
	/Users/runeimp/dev/go/runeimp/src/github.com/miekg/mmark
iMac-27:mmark runeimp$ _
```

So I figured out that there is a missing dependency or two. No mention of dependancies in the README. So I...

``` bash
iMac-27:mmark runeimp$ go get github.com/BurntSushi/toml
iMac-27:mmark runeimp$ go build
main.go:21:2: cannot find package "github.com/miekg/mmark" in any of:
	/usr/local/opt/go/libexec/src/github.com/miekg/mmark (from $GOROOT)
	/Users/runeimp/dev/go/lib/src/github.com/miekg/mmark (from $GOPATH)
	/Users/runeimp/dev/go/runeimp/src/github.com/miekg/mmark
iMac-27:mmark runeimp$ _
```

Hmm, I'm in the repo and in its _build_ directory. And yet it's complaining about missing itself as a dependancy. I'm just learning Go so lets try this from the root of the repo because I'm thinking "Maybe miekg didn't mean the actual subdirectory of the repo but the repo itself. They both have the same name, so hard to be certain."

``` bash
iMac-27:mmark runeimp$ cd ..
iMac-27:mmark runeimp$ go build
iMac-27:mmark runeimp$ _
```

Please note the _nothing_ between `go build` and the line with just the cursor. So, kind of confusing situation. Once again, love what I've learned about Go but not that much experience with it yet. "Oh, look a Makefile. Maybe miekg switched to using make tasks to simplify things."

``` bash
iMac-27:mmark runeimp$ cd ..
iMac-27:mmark runeimp$ make
( cd mmark; make)
go build -ldflags "-X main.githash=fd2f6c1" -a -tags netgo -installsuffix netgo
main.go:21:2: cannot find package "github.com/miekg/mmark" in any of:
	/usr/local/opt/go/libexec/src/github.com/miekg/mmark (from $GOROOT)
	/Users/runeimp/dev/go/lib/src/github.com/miekg/mmark (from $GOPATH)
	/Users/runeimp/dev/go/runeimp/src/github.com/miekg/mmark
make[1]: *** [mmark] Error 1
make: *** [mmark/mmark] Error 2
iMac-27:mmark runeimp$ _
```

"That error again? Why can't I just build from within the repo?" Once again, not terribly experienced with Go yet. Seems to be an issue I've seen before with Go dev in general. So I try..

``` bash
iMac-27:mmark runeimp$ go get github.com/miekg/mmark
iMac-27:mmark runeimp$ make
( cd mmark; make)
go build -ldflags "-X main.githash=fd2f6c1" -a -tags netgo -installsuffix netgo
iMac-27:mmark runeimp$ ./mmark --version
-bash: ./mmark: is a directory
iMac-27:mmark runeimp$ cd mmark
iMac-27:mmark runeimp$ ./mmark --version
1.3.6+fd2f6c1
```

Wow, that was just so not as easy as it could have been if...

1. the docs mentioned that there are two dependencies and how to resolve them.
2. the docs mentioned any command line/Go magic required to make this work for anyone. Not just the cool kids.
3. the docs mentioned the use of `make` as an option and where to find the binary after make runs.

Please accept my apologies if I'm sounding crass. I'm working on very little sleep and just want to shed some light on the potential situation and actually play with `mmake` now that I've managed to successfully create a binary.  O:-)

Also I found the repo because of `mmake` being available via Homebrew. Should probably note that in the docs as well. And any other package managers it can be installed with.

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [miekg/mmark] Build instruction..." ]
>Wow, that was just so not as easy as it could have been if...
>
>1. the docs mentioned that there are two dependencies and how to resolve them.
>2. the docs mentioned any command line/Go magic required to make this work for anyone. Not just the cool kids.
>3. the docs mentioned the use of `make` as an option and where to find the binary after make runs.
>
>Please accept my apologies if I'm sounding crass. I'm working on very little sleep and just want to shed some light on the potential situation and actually play with `mmake` now that I've managed to successfully create a binary.  O:-)
>
>Also I found the repo because of `mmake` being available via Homebrew. Should probably note that in the docs as well. And any other package managers it can be installed with.

This could/should indeed be easier... Care to send a PR to update the docs? One 
thing I do want to prevent is that it becomes a giant 
"how-to-setup-Go-on-my-computer".

**Comment by runeimp** [comment]
Yes, happy too. I've just forked the repo. Got a list of package managers you have mmark in? Did you add mmark to Homebrew or was that a _helper_?

**Comment by miekg** [comment]
I wasn't involved in the homebrew thing.
